x-pack/elastic-agent/pkg/agent/application/paths/paths.go

```diff
@@ -6,121 +6,87 @@ package paths
 
 import (
 	"flag"
+	"fmt"
 	"os"
 	"path/filepath"
-	"runtime"
-	"sync"
+	"strings"
 
-	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
 var (
-	homePath    string
-	configPath  string
-	dataPath    string
-	logsPath    string
-	serviceName string
-
-	overridesLoader sync.Once
+	topPath    string
+	configPath string
+	logsPath   string
 )
 
 func init() {
-	initialHome := initialHome()
+	topPath = initialTop()
+	configPath = topPath
+	logsPath = topPath
 
 	fs := flag.CommandLine
-	fs.StringVar(&homePath, "path.home", initialHome, "Agent root path")
-	fs.StringVar(&configPath, "path.config", initialHome, "Config path is the directory Agent looks for its config file")
-	fs.StringVar(&dataPath, "path.data", filepath.Join(initialHome, "data"), "Data path contains Agent managed binaries")
```

Comment:
What happens to the data path?

Reply:
`path.data` is no longer something the user can set, it was not using in any of the .deb or .rpm anyway. But `path.data` is now always under `${path.home}/data` this is to ensure that the upgrade process can be sure that the data lives under the home path.